File: Práctica 4 (figuras geometricas)/FigurasGeometricas.java
public class FigurasGeometricas {
    public static double area(double radio) { // Círculo
        return Math.PI * radio * radio;
    }

    double area(double base, double altura) { // Rectángulo
        return base * altura;
    }

    double area(double b, float h) { // Triángulo rectángulo
        return (b * h) / 2;
    }

    double area(double lado1, double lado2, double alto) { // Trapecio
        return ((lado1 + lado2) * alto) / 2;
    }

    double area(float lado, double apotema) { // Pentágono regular
        return (5 * lado * apotema) / 2;
    }


    public static void main(String[] args) {
        FigurasGeometricas f1 = new FigurasGeometricas();
        FigurasGeometricas f2 = new FigurasGeometricas();
        FigurasGeometricas f3 = new FigurasGeometricas();
        FigurasGeometricas f4 = new FigurasGeometricas();
        FigurasGeometricas f5 = new FigurasGeometricas();

        System.out.println("Círculo: " + f1.area(3.0)); 
        System.out.println("Rectángulo: " + f2.area(4.0, 6.0)); 
        System.out.println("Triángulo Rectángulo: " + f3.area(5.0,8)); 
        System.out.println("Trapecio: " + f4.area(6.0, 4.0, 5.0));
        System.out.println("Pentágono: " + f5.area(7, 3.0)); 
    }
}

File: Práctica 4 (figuras geometricas)/FigurasGeometricas.py
import math

class FigurasGeometricas:
    def area(self, *args):
        if len(args) == 1:
            radio = args[0]
            return math.pi * radio * radio   # circulo
        elif len(args) == 2:
            if isinstance(args[0], float) and isinstance(args[1], float):
                b, h = args
                return (b * h) / 2     # triangulo rectángulo 
            else:
                if isinstance(args[0], int) and isinstance(args[1], float):
                    lado, apotema = args
                    return ((5 * lado) * apotema) / 2   #pentagono
                else:
                    if isinstance(args[0], int) and isinstance(args[1], int):
                        b, h = args
                        return b * h    #rectángulo 
        elif len(args) == 3:
            l1, l2, h = args
            return ((l1 + l2) * h) / 2  #trapecio
        else:
            raise ValueError("Parámetro no válido")
f = FigurasGeometricas()

print("Círculo: ", f.area(3))
print("Rectángulo: ", f.area(4, 6)) 
print("Triángulo Rectángulo: ", f.area(5.0, 9.0))  
print("Trapecio: ", f.area(8, 3, 5))  
print("Pentágono: ", f.area(9, 4.5))
